src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1667,6 +1667,44 @@ class Simplex_tree {
     return sh; // None of its faces has the same filtration.
   }
 
+ public:
+  /** \brief This function resets filtration value from a given dimension. Resets all the Simplex_tree when
+   * `min_dim = 0`.
+   * `reset_filtration` may break the filtration property with `min_dim > 0`, and it is the user's responsibility to
+   * make it a valid filtration (using a large enough `filt_value`, or calling `make_filtration_non_decreasing`
+   * afterwards for instance).
+   * @param[in] filt_value The new filtration value.
+   * @param[in] min_dim The minimal dimension. Default value is 0.
+   */
+  void reset_filtration(Filtration_value filt_value, int min_dim = 0) {
+    for (auto& simplex : root_.members()) {
```

Comment:
Can't we call the recursive function on `&root` to avoid repeating this loop?

Reply:
Oh yes you are right ! Simplified on 0c843fd0